python/graphstorm/run/gsgnn_np/np_infer_gnn.py

```diff
@@ -60,19 +60,30 @@ def main(config_args):
                         model_layer_to_load=config.restore_model_layers)
     infer = GSgnnNodePredictionInferrer(model)
     infer.setup_device(device=get_device())
-    if not config.no_validation:
-        infer_idxs = infer_data.get_node_test_set(config.target_ntype)
+    # If the user requested no evaluation, ensure there's no test mask
+    if config.no_validation:
+        infer_idxs = infer_data.get_node_infer_set(config.target_ntype)
```

Comment:
For no_validation, if we have `infer_all_targets` to be True, we can also ignore the inference/test mask.